packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -269,6 +271,11 @@ export class Outbox {
 		// Let the PendingStateManager know that a message was submitted.
 		// In future, need to shift toward keeping batch as a whole!
 		for (const message of batch) {
+			if (message.deserializedContent.type === ContainerMessageType.IdAllocation) {
```

Comment:
Not sure I understand what is happening here. Can you please add some comments.
Also, what happens if it is an IdAllocation tpye message and idCompressor doesn't exist? The stashedState will be undefined, is that fine?